### PR TITLE
Alert development docker docs

### PR DESCRIPTION
**Pull request by jeffbryner**
As a follow on to #1029, update the docs and offer options for docker development of alerts and other portions of MozDef.

### CODE REVIEW

File: docker/compose/dev-sampledata.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
       context: ../../
       dockerfile: docker/compose/mozdef_sampledata/Dockerfile
     restart: always
-    command: bash -c '/opt/mozdef/envs/mozdef/examples/demo/sampleevents.sh'
+    command: bash -c 'source /opt/mozdef/envs/python/bin/activate && /opt/mozdef/envs/mozdef/examples/demo/sampleevents.sh'
```
Comment (pwnbus) [inline docker/compose/dev-sampledata.yml]: The source command is no longer needed as I've added the virtualenvironment path to the base container.

File: docker/compose/dev-rest.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
       context: ../../
       dockerfile: docker/compose/mozdef_rest/Dockerfile
     restart: always
-    command: bash -c 'python index.py -c index.conf'
+    command: bash -c 'source /opt/mozdef/envs/python/bin/activate && python index.py -c index.conf'
```
Comment (pwnbus) [inline docker/compose/dev-rest.yml]: Same here, the source command is no longer needed.

File: docs/source/alert_development_guide.rst [hunk 3]
```diff
@@ -141,15 +137,66 @@ and add your new foo alert to the others with a crontab style schedule
...
+
+
+
+To use a live, iterative environment via docker-compose:
+::
+
+  docker-compose -f docker/compose/docker-compose.yml -f docker/compose/dev-alerts.yml -p mozdef up
+
+This will start up all the containers for a mozdef environment and in addition will allow you an adhoc alerts container to work in that loads the /alerts directory as a volume in the container.
+To run the alert you are developing you will need to edit the alerts/files/config.py file as detailed above to schedule your alert. You will also need to edit it to reference the container environment as follows
```
Comment (pwnbus) [inline docs/source/alert_development_guide.rst]: The correct file to modify is actually 'alerts/lib/config.py', not 'alerts/files/config.py'.

File: docs/source/alert_development_guide.rst [hunk 4]
```diff
@@ -141,15 +137,66 @@ and add your new foo alert to the others with a crontab style schedule
...
+    'servers': ['http://elasticsearch:9200']
+  }
+
+Once you've reference the containers, you can shell into the alerts container:
+::
+  docker exec -it mozdef_alerts_1 bash
+
+Next, source the environment and start celery
+::
+  source /opt/mozdef/envs/python/bin/activate && celery -A celeryconfig worker --loglevel=info --beat
```
Comment (pwnbus) [inline docs/source/alert_development_guide.rst]: You shouldn't need to source once you're in the container anymore.

File: docs/source/alert_development_guide.rst (hunk 3 above)
Comment (jeffbryner) [reply docs/source/alert_development_guide.rst]: good catch! hence the guide ;-]

**Pull request merged**